### PR TITLE
Cast mask tensor to bool in torch.where

### LINKED ISSUE #92: Warning: where received a uint8 condition tensor
I got a warning from torch about line 249: https://github.com/kmkurn/pytorch-crf/blob/e843718540512eb04d9b756fca04a0915affe175/torchcrf/__init__.py#L249

```
UserWarning: where received a uint8 condition tensor. This behavior is deprecated and will be removed in a future version of PyTorch. Use a boolean condition instead. (Triggered internally at  ../aten/src/ATen/native/TensorCompare.cpp:328.)
```

I suppose casting the tensor to bool should fix it.

### PR DESCRIPTION
**Pull request by erickrf**
This fixes #92

### CODE REVIEW

**Comment by kmkurn** [comment]
Thanks for the PR! Looks good but I have to confirm it works for earlier PyTorch versions. This is usually done by Travis CI but since the move to travis.com the builds seem to have stopped. It'll be while before I can fix it ☹️

**Comment by kmkurn** [comment]
Hi, I've setup Github CI to replace Travis, so do you mind rebasing this PR onto the master branch?

**Comment by erickrf** [comment]
Done!

**Comment by erickrf** [comment]
Oh... I hadn't realized `tensor.bool()` wasn't available in previous versions of pytorch. By the way, the project requirements don't say anything about pytorch version, this is only set in the test config. Do you want to support pytorch 1.0?

**Comment by kmkurn** [comment]
> project requirements don't say anything about pytorch version, this is only set in the test config

Yes, this is intentional as there are several ways to install pytorch depending on the platform and if GPU support is needed. Can't really put all these in a single requirement (not sure if this has changed?).

> Do you want to support pytorch 1.0?

As pytorch is now at v1.10, perhaps dropping support for much earlier versions like v1.0 isn't a bad idea. Do you know when `Tensor.bool()` was introduced?

**Comment by myedibleenso** [comment]
>Do you know when Tensor.bool() was introduced?

@kmkurn , based off of https://github.com/pytorch/pytorch/pull/16810 and https://github.com/pytorch/pytorch/commit/aa6403bae6aeb452bd464a6ec4913d1b69548aca, it looks like it was introduced in [v1.1.0](https://github.com/pytorch/pytorch/releases/tag/v1.1.0)

**Comment by kmkurn** [comment]
Thanks @myedibleenso. The builds for PyTorch v1.1 have failed. I can't see the log any more so I'm not sure why. I can't seem to re-run the builds either ☹️
